pkg/networkpath/traceroute/utils.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package traceroute
+
+import (
+	"net"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var lookupAddrFn = net.LookupAddr
+
+// getDestinationHostname tries to convert the input destinationHost to hostname.
+// When input destinationHost is an IP, a reverse DNS call is made to convert it into a hostname.
+func getDestinationHostname(destinationHost string) string {
```

Comment:
In this first implementation, we do it for all destination.

But if it makes sense, we could optimise to exclude some IPs.

Reply:
I think you're going to find that reverse IP lookups will hold you up, so in the absence of short-ish timeouts, it would be better if we limited some IP lookups. Loopback and link-local seem like easy and obvious candidates to exclude.

Reply:
@hmahmood I agree that we need to improve the reverse IP lookup logic here. 

I think it requires some refactoring of how we handle reverse DNS for network path and might be better handled as separate task/PR. WDYT?

I created this card NETPATH-154 to handle it:

```
Problem: the current reverse DNS in traceroute / network path is naive and likely require some optimisation:

- caching the reverse reverse DNS results
- adding timeout
- excluding IPs (e.g. private? loopback?  linklocal?)
- handling/storing multiple results
```